### DISCUSSION TITLE
How are collisions handled in UnetStack's "BasicAcousticChannel"

### DISCUSSION THREAD

**Discussion by viktorlidstrom** [org-arl/unet-support#2] [Q&A] [answered]
AFAIK, this is not described in the unet handbook. When two packets collide in "BasicAcousticChannel" (overlap in time), are both lost? Are they used to adjust the SNR of each other when simulating the bit errors in "BPSKFadingModel", or is some other approach used?

**Comment by mchitre** [comment, marked as answer]
The `HalfDuplexModem` model used as the default in the simulator permits only one reception at a time. If it detects a second reception while one reception is ongoing, it generates a `CollisionNtf` for the second reception and discards it.

The `BasicAcousticChannel` tracks all the incoming receptions to compute SINR. For on ongoing reception, any new incoming receptions (even if `HalfDuplexModem` rejects them as collisions) contribute to the interference term in the SINR. Currently, the `BasicAcousticChannel` is conservative and counts any overlap in time as interference (i.e., does not reduce interference contribution for partial overlaps). When the reception ends, if the SINR is used to drive the random process that determines successful reception or otherwise. The `BasicAcousticChannel` delegates that to the `BPSKFadingModel` that uses a simple model appropriate for BPSK in a Rayleigh/Rician fading channel to decide on packet success.

**Reply by viktorlidstrom** [reply]
Perfect! Thank you for the answer, Mandar :)
